### PR TITLE
App 686 - Create a front end architecture with React Router and Redux

### PR DESCRIPTION
**Pull request by pdarde-daitan**
Create a front end architecture with React Router and Redux
https://perzoinc.atlassian.net/browse/APP-686

This first Pull Request is related to the creation of the Configurator Container as well as the "Input Description" and "Instance Table" components. The "List" and "Create" views were also created.

### CODE REVIEW

File: integration-fe/src/actions/actions.js [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+}
+
+/*
+  Creates the stream to be able to create an instance
+*/
+export function createStream(obj) {
+  // if(obj.streamType === 'IM'){
+  const streamService = SYMPHONY.services.subscribe('stream-service');
+  const _streams = [];
+  const promisedIM = streamService.createIM([7627861919706]);
```
Comment (cmarcondes-daitan) [inline integration-fe/src/actions/actions.js]: It shouldn't be fixed, use the variable.

Comment (cmarcondes-daitan) [inline integration-fe/src/actions/actions.js]: It shoudn't be fixed, change it to a variable.

File: integration-fe/src/components/TableInstance/TableInstance.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+class TableInstance extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      instances: [],
+    };
+    this.onConfigureNew = this.onConfigureNew.bind(this);
+  }
+
+  componentWillMount() {
```
Comment (cmarcondes-daitan) [inline integration-fe/src/components/TableInstance/TableInstance.jsx]: Is this really necessary? I understood that you created the instances in the state at Constructor, then you will update it at componentWillReceiveProps, am I right?

File: integration-fe/src/components/TableInstance/TableInstance.jsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+  onConfigureNew(e) {
+    e.preventDefault();
+  }
+
+  render() {
+    let dataRowObj = {};
+    return (
+      <div className='wrapper'>
+        <header>
```
Comment (cmarcondes-daitan) [inline integration-fe/src/components/TableInstance/TableInstance.jsx]: This component should be just an table component, shouldn't it? I could import just the table and use other header component. What do you think?

File: integration-fe/src/containers/CreateInstanceContainer.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import {connect} from 'react-redux';
+import {
+    CREATE_INTANCE,
+    CREATE_INSTANCE_SUCCESS,
+    ERROR,
```
Comment (cmarcondes-daitan) [inline integration-fe/src/containers/CreateInstanceContainer.jsx]: These imports have to be removed, they're never used.

**Review by cmarcondes-daitan — CHANGES_REQUESTED** [review]
Small changes requested.

File: integration-fe/src/actions/actions.js (hunk 1 above)
Comment (pdarde-daitan) [reply integration-fe/src/actions/actions.js]: Done!

File: integration-fe/src/components/TableInstance/TableInstance.jsx (hunk 2 above)
Comment (pdarde-daitan) [reply integration-fe/src/components/TableInstance/TableInstance.jsx]: Done!

File: integration-fe/src/components/TableInstance/TableInstance.jsx (hunk 3 above)
Comment (pdarde-daitan) [reply integration-fe/src/components/TableInstance/TableInstance.jsx]: Will do!

File: integration-fe/src/containers/CreateInstanceContainer.jsx (hunk 4 above)
Comment (pdarde-daitan) [reply integration-fe/src/containers/CreateInstanceContainer.jsx]: Done!

**Review by cmarcondes-daitan — APPROVED** [review]

**Pull request merged**